tests/core/test_policies.py

```diff
@@ -205,6 +220,60 @@ def create_policy(self, featurizer, priority):
         p = MappingPolicy()
         return p
 
+    @pytest.fixture(scope="module")
+    def domain_with_mapping(self):
+        return Domain.load(DEFAULT_DOMAIN_PATH_WITH_MAPPING)
+
+    @pytest.fixture
+    def tracker(self, domain_with_mapping):
+        return DialogueStateTracker(UserMessage.DEFAULT_SENDER_ID)
+
+    @pytest.fixture(
+        params=[
+            ("default", "utter_default"),
+            ("greet", "utter_greet"),
+            (USER_INTENT_RESTART, ACTION_RESTART_NAME),
+            (USER_INTENT_BACK, ACTION_BACK_NAME),
+        ]
+    )
+    def intent_mapping(self, request):
+        return request.param
+
+    def test_predict_mapped_action(self, priority, domain_with_mapping, intent_mapping):
+        policy = self.create_policy(None, priority)
+        events = [
+            ActionExecuted(ACTION_LISTEN_NAME),
+            user_uttered(intent_mapping[0], 1),
+        ]
+
+        assert (
+            self._get_next_action(policy, events, domain_with_mapping)
+            == intent_mapping[1]
+        )
+
+    def test_predict_action_listen(self, priority, domain_with_mapping, intent_mapping):
+        policy = self.create_policy(None, priority)
+        events = [
+            ActionExecuted(ACTION_LISTEN_NAME),
+            user_uttered(intent_mapping[0], 1),
+            ActionExecuted(intent_mapping[1], policy="MappingPolicy"),
```

Comment:
Not sure the `policy="MappingPolicy"` is doing what it should in this test -- I can change it to anything and the test still passes

Reply:
Hmm, I think this is due to how `_get_next_action` works. If you change the policy name the prediction returned by the policy should be all zeros, whereas if it is "MappingPolicy" it should return a unit vector that is 1 at the index of `action_listen`.